### PR TITLE
Adding volumes page to tenants

### PR DESCRIPTION
**Pull request by adfost**
Screen Shot 2021-09-09 at 9 40 20 AM
Adds a page showing volumes for a tenant.

### CODE REVIEW

File: portal-ui/src/screens/Console/Tenants/TenantDetails/VolumesSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+              elementKey: "namespace",
+              width: 90,
+            },
+            {
+              label: "Status",
+              elementKey: "status",
+              width: 120,
+            },
+            {
+              label: "Tenant",
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/VolumesSummary.tsx:148]: This column is redundant, please remove it

File: swagger-operator.yml [hunk 2]
```diff
@@ -505,6 +505,31 @@ paths:
       tags:
         - OperatorAPI
 
+  /namespaces/{namespace}/tenants/{tenant}/list-pvcs:
```
Comment (dvaldivia) [inline swagger-operator.yml:508]: the verb `GET` implies listing, and this endpoint is about the `pvc` resource, can you rename it to `/namespaces/{namespace}/tenants/{tenant}/pvcs`

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: operatorapi/operator_volumes.go [hunk 3]
```diff
@@ -83,3 +94,46 @@ func getPVCsResponse(session *models.Principal) (*models.ListPVCsResponse, *mode
...
+		return nil, prepareError(err)
+	}
+
+	// Filter Tenant PVCs. They keep their v1 tenant annotation
+	listOpts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("v1.min.io/tenant=%s", params.Tenant),
+	}
+
+	// List all PVCs
+	listAllPvcs, err2 := clientset.CoreV1().PersistentVolumeClaims("").List(ctx, listOpts)
```
Comment (dvaldivia) [inline operatorapi/operator_volumes.go:112]: you can actually filter by namespace here

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Tenants/TenantDetails/VolumesSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+  const tableActions = [
+    {
+      type: "view",
+      onClick: (record: any) => {
+        history.push(
+          `/namespaces/${record.namespace}/tenants/${record.tenant}`
+        );
+      },
+    },
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/VolumesSummary.tsx:101]: Not sure if this action is needed as you are in the tenant and only redirects you to summary page

**Review by bexsoft — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Tenants/TenantDetails/VolumesSummary.tsx (hunk 4 above)
Comment (dvaldivia) [reply portal-ui/src/screens/Console/Tenants/TenantDetails/VolumesSummary.tsx:101]: +1 this action doesn't make sense for this listing, let's remove it

**Review by dvaldivia — CHANGES_REQUESTED** [review]
Make sure you are on prettier `2.3.2`
```
prettier --version
2.3.2
```
and format your files.

**Review by bexsoft — APPROVED** [review]
LGTM

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**